src/main/java/com/scaler/lld/design/assignments/singleton/FileBasedConfigurationManagerImpl.java

```diff
@@ -2,49 +2,59 @@
 
 public class FileBasedConfigurationManagerImpl extends FileBasedConfigurationManager {
 
+    private static  FileBasedConfigurationManagerImpl INSTANCE = null;
+
+    private FileBasedConfigurationManagerImpl() {
+        super();
+    }
+
     @Override
     public String getConfiguration(String key) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+       return  properties.getProperty(key);
     }
 
     @Override
     public <T> T getConfiguration(String key, Class<T> type) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'getConfiguration'");
+        return convert(properties.getProperty(key), type);
```

Comment:
An edge-case you should handle here is the property being null.
Try to use the `Optional` class instead of an if-else block.